lib/api/objectGet.js

```diff
@@ -156,12 +156,21 @@ function objectGet(authInfo, request, returnTagCount, log, callback) {
                     targetLocation, objMD.replicationInfo);
                 if (repBackendResult.error) {
                     log.error('Error with location constraint header', {
+                        bucketName, objectKey, versionId,
                         error: repBackendResult.error,
                         status: repBackendResult.status,
                     });
                     return callback(repBackendResult.error, null, corsHeaders);
                 }
-                dataLocator = repBackendResult.dataLocator;
+                const targetDataLocator = repBackendResult.dataLocator;
+                if (targetDataLocator) {
+                    dataLocator = targetDataLocator;
+                } else {
+                    log.info('Doing GET on source location', {
```

Comment:
I had though it would be useful to have an info log because of the possible "race condition on read" known issue, it may help tracking those issues. But checking the request timestamps in cloudserver and backbeat should be enough to identify if a read error was caused by this race, so I can make this a debug log.